softlayer/resource_softlayer_global_ip.go

```diff
@@ -0,0 +1,265 @@
+        return resourceSoftLayerGlobalIpUpdate(d, meta)
+
+}
+
+func resourceSoftLayerGlobalIpRead(d *schema.ResourceData, meta interface{}) error {
+        sess := meta.(*session.Session)
+        service := services.GetNetworkSubnetIpAddressGlobalService(sess)
+
+        globalIpId, err := strconv.Atoi(d.Id())
+        if err != nil {
+                return fmt.Errorf("Not a valid global ip ID, must be an integer: %s", err)
+        }
+
+        globalIp, err := service.Id(globalIpId).Mask(GlobalIpMask).GetObject()
+        if err != nil {
+                if strings.Contains(err.Error(), "404 Not Found") {
+                        d.SetId("")
+                        return nil
+                }
+
+                return fmt.Errorf("Error retrieving Global Ip: %s", err)
+        }
+
+        d.Set("id", *globalIp.Id)
+        d.Set("ip_address", *globalIp.IpAddress.IpAddress)
+        d.Set("routes_to", *globalIp.DestinationIpAddress.IpAddress)        
+
+        return nil
+}
+
+func resourceSoftLayerGlobalIpUpdate(d *schema.ResourceData, meta interface{}) error {
+        sess := meta.(*session.Session)
+        service := services.GetNetworkSubnetIpAddressGlobalService(sess)
+
+        globalIpId, err := strconv.Atoi(d.Id())
+        if err != nil {
+                return fmt.Errorf("Not a valid global ip ID, must be an integer: %s", err)
+        }
+        
+        if d.HasChange("routes_to") {
+                _, err = service.Id(globalIpId).Route(sl.String(d.Get("routes_to").(string)))
+        }        
+
+        if err != nil {
+                return fmt.Errorf("Error editing Global Ip: %s", err)
+        }
+        return resourceSoftLayerGlobalIpRead(d, meta)
+}
+
+func resourceSoftLayerGlobalIpDelete(d *schema.ResourceData, meta interface{}) error {
+        sess := meta.(*session.Session)
+        service := services.GetNetworkSubnetIpAddressGlobalService(sess)
+
+        globalIpId, err := strconv.Atoi(d.Id())
+        if err != nil {
+                return fmt.Errorf("Not a valid global ip ID, must be an integer: %s", err)
+        }
+
+        billingItem, err := service.Id(globalIpId).GetBillingItem()
+        if err != nil {
+                return fmt.Errorf("Error deleting global ip: %s", err)
+        }
+
+        if billingItem.Id == nil {
+                return nil
+        }
+
+        _, err = services.GetBillingItemService(sess).Id(*billingItem.Id).CancelService()
+
+        return err
+}
+
+func resourceSoftLayerGlobalIpExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+        sess := meta.(*session.Session)
+        service := services.GetNetworkSubnetIpAddressGlobalService(sess)
+
+        globalIpId, err := strconv.Atoi(d.Id())
+        if err != nil {
+                return false, fmt.Errorf("Not a valid ID, must be an integer: %s", err)
+        }
+
+        result, err := service.Id(globalIpId).GetObject()
+        return result.Id != nil && err == nil && *result.Id == globalIpId, nil
```

Comment:
@srinivasdandibhotla The problem with the above is that, *if* there is an error, we are suppressing it by returning `nil` for the second return value.

Instead, when there is an error preventing a conclusive existence check, we need to report it to Terraform by returning that error (optionally wrapped), so that Terraform will not incorrectly mark an active resource as destroyed, or conversely, report a destroyed resource as active.  So we need to separate the error check and existence check

We have an issue in our internal tracker to address this for other resources.  Let's use the correct way for this new resource.